### PR TITLE
Enable network service in RHEL8

### LINKED ISSUE #688: Rebooting the RHEL8 helper makes it unreachable
**Describe the bug**
After configuring the ifcfg-* files and rebooting the helper, it is unreachable because the network service is not enabled in RHEL8

**To Reproduce**
Run the 02 script and reboot the helper node.

**Expected/observed behavior**
Cannot log-in via network and `ip a` shows no IPs

**Additional context**

### PR DESCRIPTION
**Pull request by e-minguez**
It is disabled by default in RHEL8
Fixes #688

### CODE REVIEW

File: 02_configure_host.sh [hunk 1]
```diff
@@ -111,6 +111,9 @@ if [ "$MANAGE_INT_BRIDGE" == "y" ]; then
         else
            sudo systemctl restart network
         fi
+        if [ "${RHEL8}" = "True" ] ; then
+          sudo systemctl enable network
+        fi
```
Comment (hardys) [inline 02_configure_host.sh]: I'm wondering how this ends up disabled?  Was it a manual ISO install where no network was activated as part of the install?

Regardless this seems unrelated to the OS version, perhaps we should just unconditionally enable then restart it, to remove the restart duplication above?

Comment (hardys) [reply 02_configure_host.sh]: Ah I see the restart de-duplication is handled in https://github.com/openshift-metal3/dev-scripts/pull/690 thanks!

Comment (e-minguez) [reply 02_configure_host.sh]: I've added the enable only if the ifcfg-* files are modified

Comment (russellb) [reply 02_configure_host.sh]: Is the `network` service only for handling `ifcfg` scripts?  In RHEL8 that's not installed by default.  We install it so we can still use `ifcfg` scripts.  That's probably why it has to be enabled in RHEL8 and we didn't see that before.

File: 02_configure_host.sh [hunk 2]
```diff
@@ -112,6 +112,12 @@ if [ "$MANAGE_INT_BRIDGE" == "y" ]; then
     fi
 fi
 
+# If there were modifications to the /etc/sysconfig/network-scripts/ifcfg-*
+# files, it is required to enable the network service
+if [ "$MANAGE_INT_BRIDGE" == "y" ] || [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
+  sudo systemctl enable network
```
Comment (russellb) [inline 02_configure_host.sh]: I suspect this was only needed for RHEL8 since we install `network-scripts` for `ifcfg` file support, which isn't there by default.  I suppose this is harmless elsewhere though.

**Comment by hardys** [comment]
This is fine, but I can't help wondering if we should just move to nmcli for RHEL8 instead?

Here's an example of configuring an interface with nmcli:

https://github.com/hardys/installer/blob/issue_2060/data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template#L29

Not blocking this as a quick-fix, but wanted to have that discussion and now seems a good time to mention it :)

**Comment by russellb** [comment]
+1 to moving to network manager for all of this!


On Jul 24, 2019, at 11:01 AM, Steven Hardy <notifications@github.com> wrote:

This is fine, but I can't help wondering if we should just move to nmcli
for RHEL8 instead?

Here's an example of configuring an interface with nmcli:

https://github.com/hardys/installer/blob/issue_2060/data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template#L29

Not blocking this as a quick-fix, but wanted to have that discussion and
now seems a good time to mention it :)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/openshift-metal3/dev-scripts/pull/689?email_source=notifications&email_token=AACLQCQ3JUN7RHPEXDBYK7TQBBVCZA5CNFSM4IGNF4S2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2WTXVI#issuecomment-514669525>,
or mute the thread

**Comment by e-minguez** [comment]
Can we have this merged then create an issue to track the NetworkManager modifications? Thanks!

**Comment by hardys** [comment]
> Can we have this merged then create an issue to track the NetworkManager modifications? Thanks!
> 
> (also, TIL about https://github.com/nmstate/nmstate)

Sure, and yes maybe nmstate would be a nice thing to investigate as an alternative to shell hacking :)
